src/preset_cli/cli/superset/sync/dbt/metrics.py

```diff
@@ -89,32 +89,27 @@ def get_metric_expression(metric_name: str, metrics: Dict[str, MetricSchema]) ->
         return f"COUNT(DISTINCT {sql})"
 
     if type_ in {"expression", "derived"}:
+        if metric.get("skip_parsing"):
+            return remove_line_break_wrappers(sql)
+
         try:
             expression = sqlglot.parse_one(sql, dialect=metric["dialect"])
+            tokens = expression.find_all(exp.Column)
+
+            for token in tokens:
+                if token.sql() in metrics:
+                    parent_sql = get_metric_expression(token.sql(), metrics)
+                    parent_expression = sqlglot.parse_one(
+                        parent_sql,
+                        dialect=metric["dialect"],
+                    )
+                    token.replace(parent_expression)
+
+            return expression.sql(dialect=metric["dialect"])
```

Comment:
Haven't really changed anything in this block, just moved it inside the `try/except` block as the `sqlglot.parse_one()` call in the `for` loop might also fail.